### PR TITLE
PHSC138 Keymap for Atom47

### PR DESCRIPTION
**Pull request by PHSC138**
Adding a keymap for the atom47

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/evyd13/atom47/keymaps/PHSC138/config.h [hunk 1]
```diff
@@ -0,0 +1,2 @@
+#pragma once
```
Comment (drashna) [inline keyboards/evyd13/atom47/keymaps/PHSC138/config.h:1]: Could you add a GPL2+ compatible license header here?  All of the source files need to have a license explicitly listed.

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

File: keyboards/evyd13/atom47/keymaps/PHSC138/README.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
+# PHSC138's Layout for the atom47
```
Comment (drashna) [inline keyboards/evyd13/atom47/keymaps/PHSC138/README.md:1]: please rename this file and folder to be lower case.

File: keyboards/evyd13/atom47/keymaps/PHSC138/keymap.c [hunk 3]
```diff
@@ -0,0 +1,389 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
+#include "config.h"
```
Comment (drashna) [inline keyboards/evyd13/atom47/keymaps/PHSC138/keymap.c:18]:
```suggestion
```

File: keyboards/evyd13/atom47/keymaps/PHSC138/keymap.c [hunk 4]
```diff
@@ -0,0 +1,389 @@
...
+
+// These are all aliases for the function layers.
+#define _BASE 0
+#define _PROG 1
+#define _GAME 2
+#define _FN   7
+#define _FN1  8
+#define _NUM  9
```
Comment (drashna) [inline keyboards/evyd13/atom47/keymaps/PHSC138/keymap.c:26]: Enums are much simpler to handle, especially the skip from 2 to 7 still populates those inbetween layers, and can increase the finished compile size.
```suggestion
enum custom_layers {
    _BASE,
    _PROG,
    _GAME,
    _FN,
    _FN1,
    _NUM
};
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/evyd13/atom47/keymaps/PHSC138/README.md (hunk 2 above)
Comment (PHSC138) [reply keyboards/evyd13/atom47/keymaps/PHSC138/README.md]: Resolved here [b1e0652](https://github.com/qmk/qmk_firmware/pull/18768/commits/b1e065224b86b31be3432a7808fbfa6a0d9f985a)

File: keyboards/evyd13/atom47/keymaps/PHSC138/config.h (hunk 1 above)
Comment (PHSC138) [reply keyboards/evyd13/atom47/keymaps/PHSC138/config.h]: Resolved here [5c04134](https://github.com/qmk/qmk_firmware/pull/18768/commits/5c041345e76837e83a89a2d732937392c0d1e8e2)

File: keyboards/evyd13/atom47/keymaps/PHSC138/keymap.c (hunk 3 above)
Comment (PHSC138) [reply keyboards/evyd13/atom47/keymaps/PHSC138/keymap.c]: Resolved here [5c04134](https://github.com/qmk/qmk_firmware/pull/18768/commits/5c041345e76837e83a89a2d732937392c0d1e8e2)

File: keyboards/evyd13/atom47/keymaps/PHSC138/keymap.c (hunk 4 above)
Comment (PHSC138) [reply keyboards/evyd13/atom47/keymaps/PHSC138/keymap.c]: Resolved here [5c04134](https://github.com/qmk/qmk_firmware/pull/18768/commits/5c041345e76837e83a89a2d732937392c0d1e8e2)

**Review by drashna — APPROVED** [review]

File: keyboards/evyd13/atom47/keymaps/phsc138/rules.mk [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+MOUSEKEY_ENABLE = no        # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+NKRO_ENABLE = yes           # Enable N-Key Rollover
+RGBLIGHT_ENABLE = no        # Enable keyboard underlight functionality (+4870)
+AUDIO_ENABLE = no           # Audio output
+UNICODE_ENABLE  = no        # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+TAP_DANCE_ENABLE = yes      # Enable single keys to have multiple functions
```
Comment (fauxpark) [inline keyboards/evyd13/atom47/keymaps/phsc138/rules.mk:14]:
```suggestion
MOUSEKEY_ENABLE = no
NKRO_ENABLE = yes
TAP_DANCE_ENABLE = yes
```

File: keyboards/evyd13/atom47/keymaps/phsc138/readme.md [hunk 6]
```diff
@@ -0,0 +1,71 @@
+# PHSC138's Layout for the atom47
+## Make
+```sh
+qmk make -kb evyd13/atom47/rev3 -km PHSC138
+```
+## Flash
+```sh
+qmk flash -kb evyd13/atom47/rev3 -km PHSC138
+```
```
Comment (fauxpark) [inline keyboards/evyd13/atom47/keymaps/phsc138/readme.md:9]:
```suggestion

```
QMK docs already explain how to do this.

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

File: keyboards/evyd13/atom47/keymaps/phsc138/keymap.c [hunk 7]
```diff
@@ -0,0 +1,390 @@
...
+        // _FN
+        // _FN1
+        // _NUM
+        // _GAME
+    // _GAME
+    // _BASE
+
+// Tap pn for toggle to _PROG, or hold for numpad
+[_BASE] = LAYOUT_split_space(
+    KC_GESC,  KC_Q,     KC_W,     KC_E,        KC_R,        KC_T,       KC_Y,    KC_U,      KC_I,    KC_O,       KC_P,     KC_DEL,   KC_BSPC,
```
Comment (drashna) [inline keyboards/evyd13/atom47/keymaps/phsc138/keymap.c:83]:
```suggestion
    QK_GESC,  KC_Q,     KC_W,     KC_E,        KC_R,        KC_T,       KC_Y,    KC_U,      KC_I,    KC_O,       KC_P,     KC_DEL,   KC_BSPC,
```

File: keyboards/evyd13/atom47/keymaps/phsc138/readme.md [hunk 8]
```diff
@@ -0,0 +1,63 @@
+# PHSC138's Layout for the atom47
+## Layers
+<!-- TODO: pictures -->
+```c
+// Tap pn for toggle to _PROG, or hold for numpad
+[_BASE] = LAYOUT_split_space(
+    KC_GESC,  KC_Q,     KC_W,     KC_E,        KC_R,        KC_T,       KC_Y,    KC_U,      KC_I,    KC_O,       KC_P,     KC_DEL,   KC_BSPC,
```
Comment (drashna) [inline keyboards/evyd13/atom47/keymaps/phsc138/readme.md:7]:
```suggestion
    QK_GESC,  KC_Q,     KC_W,     KC_E,        KC_R,        KC_T,       KC_Y,    KC_U,      KC_I,    KC_O,       KC_P,     KC_DEL,   KC_BSPC,
```

Comment (PHSC138) [reply keyboards/evyd13/atom47/keymaps/phsc138/readme.md:7]: KC_GESC -> QK_GESC

File: keyboards/evyd13/atom47/keymaps/phsc138/readme.md [hunk 9]
```diff
@@ -0,0 +1,63 @@
+# PHSC138's Layout for the atom47
+## Layers
+<!-- TODO: pictures -->
+```c
```
Comment (fauxpark) [inline keyboards/evyd13/atom47/keymaps/phsc138/readme.md:4]: You don't really need to duplicate the entire keymap in the readme...

File: keyboards/evyd13/atom47/keymaps/phsc138/keymap.c [hunk 10]
```diff
@@ -0,0 +1,390 @@
...
+ *  Not a key that is hit frequently in a sentence
+ *  Not a key that is used frequently to double tap, for example 'tab' is often double tapped in a terminal, or
+ *    in a web form. So 'tab' would be a poor choice for a tap dance.
+ *  Letters used in common words as a double. For example 'p' in 'pepper'. If a tap dance function existed on the
+ *    letter 'p', the word 'pepper' would be quite frustating to type.
+ *
+ * For the third point, there does exist the 'DOUBLE_SINGLE_TAP', however this is not fully tested
+ *
+ */
+int cur_dance(qk_tap_dance_state_t *state) {
```
Comment (drashna) [inline keyboards/evyd13/atom47/keymaps/phsc138/keymap.c:175]:
```suggestion
uint8_t cur_dance(tap_dance_state_t *state) {
```

File: keyboards/evyd13/atom47/keymaps/phsc138/keymap.c [hunk 11]
```diff
@@ -0,0 +1,390 @@
...
+    else return 8; //magic number. At some point this method will expand to work for more presses
+}
+
+//instanalize an instance of 'tap' for the 'pn' tap dance.
+static tap pn_tap_state = {
+    .is_press_action = 1,
+    .state = 0
+};
+
+void pn_finished(qk_tap_dance_state_t *state, void *user_data) {
```
Comment (drashna) [inline keyboards/evyd13/atom47/keymaps/phsc138/keymap.c:206]:
```suggestion
void pn_finished(tap_dance_state_t *state, void *user_data) {
```

File: keyboards/evyd13/atom47/keymaps/phsc138/keymap.c [hunk 12]
```diff
@@ -0,0 +1,390 @@
...
+        //case DOUBLE_TAP: register_code(KC_ESC); break;
+        //case DOUBLE_HOLD: register_code(KC_LALT); break;
+        //case DOUBLE_SINGLE_TAP: register_code(KC_X); unregister_code(KC_X); register_code(KC_X);
+        //Last case is for fast typing. Assuming your key is `f`:
+        //For example, when typing the word `buffer`, and you want to make sure that you send `ff` and not `Esc`.
+        //In order to type `ff` when typing fast, the next character will have to be hit within the `TAPPING_TERM`, which by default is 200ms.
+    }
+}
+
+void pn_reset(qk_tap_dance_state_t *state, void *user_data) {
```
Comment (drashna) [inline keyboards/evyd13/atom47/keymaps/phsc138/keymap.c:220]:
```suggestion
void pn_reset(tap_dance_state_t *state, void *user_data) {
```

File: keyboards/evyd13/atom47/keymaps/phsc138/keymap.c [hunk 13]
```diff
@@ -0,0 +1,390 @@
...
+    }
+    pn_tap_state.state = 0;
+}
+
+static tap lalt_tap_state = {
+    .is_press_action = 1,
+    .state = 0
+};
+
+void lalt_finished(qk_tap_dance_state_t *state, void *user_data) {
```
Comment (drashna) [inline keyboards/evyd13/atom47/keymaps/phsc138/keymap.c:236]:
```suggestion
void lalt_finished(tap_dance_state_t *state, void *user_data) {
```

File: keyboards/evyd13/atom47/keymaps/phsc138/keymap.c [hunk 14]
```diff
@@ -0,0 +1,390 @@
...
+    lalt_tap_state.state = cur_dance(state);
+    switch(lalt_tap_state.state) {
+        case SINGLE_TAP: register_code(KC_LBRC); break;
+        case SINGLE_HOLD: register_code(KC_LALT); break;
+    }
+}
+
+void lalt_reset(qk_tap_dance_state_t *state, void *user_data) {
```
Comment (drashna) [inline keyboards/evyd13/atom47/keymaps/phsc138/keymap.c:244]:
```suggestion
void lalt_reset(tap_dance_state_t *state, void *user_data) {
```

File: keyboards/evyd13/atom47/keymaps/phsc138/keymap.c [hunk 15]
```diff
@@ -0,0 +1,390 @@
...
+    }
+    lalt_tap_state.state = 0;
+}
+
+static tap ralt_tap_state = {
+    .is_press_action = 1,
+    .state = 0
+};
+
+void ralt_finished(qk_tap_dance_state_t *state, void *user_data) {
```
Comment (drashna) [inline keyboards/evyd13/atom47/keymaps/phsc138/keymap.c:257]:
```suggestion
void ralt_finished(tap_dance_state_t *state, void *user_data) {
```

File: keyboards/evyd13/atom47/keymaps/phsc138/keymap.c [hunk 16]
```diff
@@ -0,0 +1,390 @@
...
+    ralt_tap_state.state = cur_dance(state);
+    switch(ralt_tap_state.state) {
+        case SINGLE_TAP: register_code(KC_RBRC); break;
+        case SINGLE_HOLD: register_code(KC_RALT); break;
+    }
+}
+
+void ralt_reset(qk_tap_dance_state_t *state, void *user_data) {
```
Comment (drashna) [inline keyboards/evyd13/atom47/keymaps/phsc138/keymap.c:265]:
```suggestion
void ralt_reset(tap_dance_state_t *state, void *user_data) {
```

File: keyboards/evyd13/atom47/keymaps/phsc138/keymap.c [hunk 17]
```diff
@@ -0,0 +1,390 @@
...
+    }
+    ralt_tap_state.state = 0;
+}
+
+static tap rctl_tap_state = {
+    .is_press_action = 1,
+    .state = 0
+};
+
+void rctl_finished(qk_tap_dance_state_t *state, void *user_data) {
```
Comment (drashna) [inline keyboards/evyd13/atom47/keymaps/phsc138/keymap.c:278]:
```suggestion
void rctl_finished(tap_dance_state_t *state, void *user_data) {
```

File: keyboards/evyd13/atom47/keymaps/phsc138/keymap.c [hunk 18]
```diff
@@ -0,0 +1,390 @@
...
+    rctl_tap_state.state = cur_dance(state);
+    switch(rctl_tap_state.state) {
+        case SINGLE_TAP: register_code(KC_RSFT); register_code(KC_RBRC); break;
+        case SINGLE_HOLD: register_code(KC_RCTL); break;
+    }
+}
+
+void rctl_reset(qk_tap_dance_state_t *state, void *user_data) {
```
Comment (drashna) [inline keyboards/evyd13/atom47/keymaps/phsc138/keymap.c:286]:
```suggestion
void rctl_reset(tap_dance_state_t *state, void *user_data) {
```

File: keyboards/evyd13/atom47/keymaps/phsc138/keymap.c [hunk 19]
```diff
@@ -0,0 +1,390 @@
...
+    }
+    rctl_tap_state.state = 0;
+}
+
+static tap lctl_tap_state = {
+    .is_press_action = 1,
+    .state = 0
+};
+
+void lctl_finished(qk_tap_dance_state_t *state, void *user_data) {
```
Comment (drashna) [inline keyboards/evyd13/atom47/keymaps/phsc138/keymap.c:299]:
```suggestion
void lctl_finished(tap_dance_state_t *state, void *user_data) {
```

File: keyboards/evyd13/atom47/keymaps/phsc138/keymap.c [hunk 20]
```diff
@@ -0,0 +1,390 @@
...
+    lctl_tap_state.state = cur_dance(state);
+    switch(lctl_tap_state.state) {
+        case SINGLE_TAP: register_code(KC_LSFT); register_code(KC_LBRC); break;
+        case SINGLE_HOLD: register_code(KC_LCTL); break;
+    }
+}
+
+void lctl_reset(qk_tap_dance_state_t *state, void *user_data) {
```
Comment (drashna) [inline keyboards/evyd13/atom47/keymaps/phsc138/keymap.c:307]:
```suggestion
void lctl_reset(tap_dance_state_t *state, void *user_data) {
```

File: keyboards/evyd13/atom47/keymaps/phsc138/keymap.c [hunk 21]
```diff
@@ -0,0 +1,390 @@
...
+
+static tap d20_tap_state = {
+    .is_press_action = 1,
+    .state = 0,
+};
+
+int d20_srand = 0;
+uint32_t timer_seed;
+
+void d20_finished(qk_tap_dance_state_t *state, void *user_data) {
```
Comment (drashna) [inline keyboards/evyd13/atom47/keymaps/phsc138/keymap.c:323]:
```suggestion
void d20_finished(tap_dance_state_t *state, void *user_data) {
```

File: keyboards/evyd13/atom47/keymaps/phsc138/keymap.c [hunk 22]
```diff
@@ -0,0 +1,390 @@
...
+            res[1] = (char)(roll % 10 + '0');
+
+            send_string(res);
+
+            break;
+                         }
+    }
+}
+
+void d20_reset(qk_tap_dance_state_t *state, void *user_data) {
```
Comment (drashna) [inline keyboards/evyd13/atom47/keymaps/phsc138/keymap.c:379]:
```suggestion
void d20_reset(tap_dance_state_t *state, void *user_data) {
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/evyd13/atom47/keymaps/phsc138/readme.md (hunk 9 above)
Comment (PHSC138) [reply keyboards/evyd13/atom47/keymaps/phsc138/readme.md]: I was planning on adding some nice pictures here, finally got around to it haha

File: keyboards/evyd13/atom47/keymaps/phsc138/keymap.c (hunk 22 above)
Comment (PHSC138) [reply keyboards/evyd13/atom47/keymaps/phsc138/keymap.c]: Fixed this comment here: [d9e04e8](https://github.com/qmk/qmk_firmware/pull/18768/commits/d9e04e80af7200159a9307be8ed3559a94052311)

File: keyboards/evyd13/atom47/keymaps/phsc138/keymap.c (hunk 21 above)
Comment (PHSC138) [reply keyboards/evyd13/atom47/keymaps/phsc138/keymap.c]: Fixed this comment here: [d9e04e8](https://github.com/qmk/qmk_firmware/pull/18768/commits/d9e04e80af7200159a9307be8ed3559a94052311)

File: keyboards/evyd13/atom47/keymaps/phsc138/keymap.c (hunk 20 above)
Comment (PHSC138) [reply keyboards/evyd13/atom47/keymaps/phsc138/keymap.c]: Fixed this comment here: [d9e04e8](https://github.com/qmk/qmk_firmware/pull/18768/commits/d9e04e80af7200159a9307be8ed3559a94052311)

File: keyboards/evyd13/atom47/keymaps/phsc138/keymap.c (hunk 19 above)
Comment (PHSC138) [reply keyboards/evyd13/atom47/keymaps/phsc138/keymap.c]: Fixed this comment here: [d9e04e8](https://github.com/qmk/qmk_firmware/pull/18768/commits/d9e04e80af7200159a9307be8ed3559a94052311)

File: keyboards/evyd13/atom47/keymaps/phsc138/keymap.c (hunk 18 above)
Comment (PHSC138) [reply keyboards/evyd13/atom47/keymaps/phsc138/keymap.c]: Fixed this comment here: [d9e04e8](https://github.com/qmk/qmk_firmware/pull/18768/commits/d9e04e80af7200159a9307be8ed3559a94052311)

File: keyboards/evyd13/atom47/keymaps/phsc138/keymap.c (hunk 14 above)
Comment (PHSC138) [reply keyboards/evyd13/atom47/keymaps/phsc138/keymap.c]: Fixed this comment here: [d9e04e8](https://github.com/qmk/qmk_firmware/pull/18768/commits/d9e04e80af7200159a9307be8ed3559a94052311)

File: keyboards/evyd13/atom47/keymaps/phsc138/keymap.c (hunk 13 above)
Comment (PHSC138) [reply keyboards/evyd13/atom47/keymaps/phsc138/keymap.c]: Fixed this comment here: [d9e04e8](https://github.com/qmk/qmk_firmware/pull/18768/commits/d9e04e80af7200159a9307be8ed3559a94052311)

File: keyboards/evyd13/atom47/keymaps/phsc138/keymap.c (hunk 12 above)
Comment (PHSC138) [reply keyboards/evyd13/atom47/keymaps/phsc138/keymap.c]: Fixed this comment here: [d9e04e8](https://github.com/qmk/qmk_firmware/pull/18768/commits/d9e04e80af7200159a9307be8ed3559a94052311)

File: keyboards/evyd13/atom47/keymaps/phsc138/keymap.c (hunk 11 above)
Comment (PHSC138) [reply keyboards/evyd13/atom47/keymaps/phsc138/keymap.c]: Fixed this comment here: [d9e04e8](https://github.com/qmk/qmk_firmware/pull/18768/commits/d9e04e80af7200159a9307be8ed3559a94052311)

File: keyboards/evyd13/atom47/keymaps/phsc138/keymap.c (hunk 10 above)
Comment (PHSC138) [reply keyboards/evyd13/atom47/keymaps/phsc138/keymap.c]: Fixed this comment here: [d9e04e8](https://github.com/qmk/qmk_firmware/pull/18768/commits/d9e04e80af7200159a9307be8ed3559a94052311)
and here: [1586718](https://github.com/qmk/qmk_firmware/pull/18768/commits/158671850f97100d384107d0cfdac0d4a9ee48f5)

File: keyboards/evyd13/atom47/keymaps/phsc138/vortex-core-base.png
Comment (drashna) [inline keyboards/evyd13/atom47/keymaps/phsc138/vortex-core-base.png:1]: This file needs to be removed.

Comment (PHSC138) [reply keyboards/evyd13/atom47/keymaps/phsc138/vortex-core-base.png:1]: Removed!
[daf24cd](https://github.com/qmk/qmk_firmware/pull/18768/commits/daf24cdf3757f93cdb2592b5ca7ac138a1b8334b)

File: keyboards/evyd13/atom47/keymaps/phsc138/keymap.c [hunk 23]
```diff
@@ -0,0 +1,390 @@
...
+    PN_SWAP = 0,
+    LAPO = 1,
+    LCPO = 2,
+    RAPC = 3,
+    RCPC = 4,
+    D20 = 5
+};
+
+enum custom_keycodes {
+    PROF_MAC = SAFE_RANGE
```
Comment (drashna) [inline keyboards/evyd13/atom47/keymaps/phsc138/keymap.c:57]:
```suggestion
    PROF_MAC = QK_USER,
```

Comment (PHSC138) [reply keyboards/evyd13/atom47/keymaps/phsc138/keymap.c:57]: This seems trivial?
`#define SAFE_RANGE             QK_USER`

Comment (fauxpark) [reply keyboards/evyd13/atom47/keymaps/phsc138/keymap.c:57]: `SAFE_RANGE` will eventually be removed, that's why that define is there.

Comment (PHSC138) [reply keyboards/evyd13/atom47/keymaps/phsc138/keymap.c]: Thanks for the explanation

**Review by drashna — APPROVED** [review]

**Pull request merged**